runtime/common/src/parachains.rs

```diff
@@ -304,6 +532,46 @@ decl_module! {
 			Ok(())
 		}
 
+		/// Provide a proof that some validator has commited a double-vote.
+		///
+		/// The weight is 0; in order to avoid DoS a `SignedExtension` validation
+		/// is implemented.
+		#[weight = SimpleDispatchInfo::FixedNormal(0)]
+		pub fn report_double_vote(
+			origin,
+			report: DoubleVoteReport,
+			key_owner_proof: Vec<u8>,
```

Comment:
I'll comment here to help avoid some trouble with generics.
The `key_owner_proof` should definitely be a member of the `DoubleVoteReport`.
The `KeyOwnerProofSystem` should be an associated type of `trait Trait`.
`DoubleVoteReport` should become `DoubleVoteReport<Proof>`, which in this method will be instantiated as `DoubleVoteReport<<T::KeyOwnerProofSystem as KeyOwnerProofSystem>::Proof>`.